src/protocols/secure_channel/tests/TestStatusReport.cpp

```diff
@@ -73,121 +82,57 @@ void TestStatusReport_WithData(nlTestSuite * inSuite, void * inContext)
-    NL_TEST_ASSERT(inSuite, reportToParse.GetProtocolId() == protocolId);
-    NL_TEST_ASSERT(inSuite, reportToParse.GetProtocolCode() == protocolCode);
+    EXPECT_EQ(reportToParse.Parse(std::move(msgBuf)), CHIP_NO_ERROR);
+    EXPECT_EQ(reportToParse.GetGeneralCode(), generalCode);
+    EXPECT_EQ(reportToParse.GetProtocolId(), protocolId);
+    EXPECT_EQ(reportToParse.GetProtocolCode(), protocolCode);
 
     const System::PacketBufferHandle & rcvData = reportToParse.GetProtocolData();
-    if (rcvData.IsNull())
-    {
-        NL_TEST_ASSERT(inSuite, false);
-        return;
-    }
-    NL_TEST_ASSERT(inSuite, rcvData->DataLength() == dataLen);
-    NL_TEST_ASSERT(inSuite, !memcmp(rcvData->Start(), data, dataLen));
+    EXPECT_FALSE(rcvData.IsNull());
+    EXPECT_EQ(rcvData->DataLength(), dataLen);
+    EXPECT_EQ(memcmp(rcvData->Start(), data, dataLen), 0);
 }
 
-void TestBadStatusReport(nlTestSuite * inSuite, void * inContext)
+TEST_F(TestStatusReport, TestBadStatusReport)
 {
     StatusReport report;
     System::PacketBufferHandle badMsg = System::PacketBufferHandle::New(10);
     CHIP_ERROR err                    = report.Parse(std::move(badMsg));
-    NL_TEST_ASSERT(inSuite, err != CHIP_NO_ERROR);
+    EXPECT_NE(err, CHIP_NO_ERROR);
 
     StatusReport report2;
     badMsg = nullptr;
     err    = report2.Parse(std::move(badMsg));
-    NL_TEST_ASSERT(inSuite, err != CHIP_NO_ERROR);
+    EXPECT_NE(err, CHIP_NO_ERROR);
 }
 
-void TestMakeBusyStatusReport(nlTestSuite * inSuite, void * inContext)
+TEST_F(TestStatusReport, TestMakeBusyStatusReport)
 {
     GeneralStatusCode generalCode                 = GeneralStatusCode::kBusy;
     auto protocolId                               = SecureChannel::Id;
     uint16_t protocolCode                         = kProtocolCodeBusy;
     System::Clock::Milliseconds16 minimumWaitTime = System::Clock::Milliseconds16(5000);
 
     System::PacketBufferHandle handle = StatusReport::MakeBusyStatusReportMessage(minimumWaitTime);
-    NL_TEST_ASSERT(inSuite, !handle.IsNull());
+    EXPECT_FALSE(handle.IsNull());
 
     StatusReport reportToParse;
-    CHIP_ERROR err = reportToParse.Parse(std::move(handle));
-    NL_TEST_ASSERT(inSuite, CHIP_NO_ERROR == err);
-    NL_TEST_ASSERT(inSuite, reportToParse.GetGeneralCode() == generalCode);
-    NL_TEST_ASSERT(inSuite, reportToParse.GetProtocolId() == protocolId);
-    NL_TEST_ASSERT(inSuite, reportToParse.GetProtocolCode() == protocolCode);
+    EXPECT_EQ(reportToParse.Parse(std::move(handle)), CHIP_NO_ERROR);
+    EXPECT_EQ(reportToParse.GetGeneralCode(), generalCode);
+    EXPECT_EQ(reportToParse.GetProtocolId(), protocolId);
+    EXPECT_EQ(reportToParse.GetProtocolCode(), protocolCode);
 
     const System::PacketBufferHandle & rcvData = reportToParse.GetProtocolData();
-    NL_TEST_ASSERT(inSuite, !rcvData.IsNull());
-    NL_TEST_ASSERT(inSuite, rcvData->DataLength() == sizeof(minimumWaitTime));
+    EXPECT_FALSE(rcvData.IsNull());
```

Comment:
```suggestion
    ASSERT_FALSE(rcvData.IsNull());
```
Since we're about to dereference rcvData